[Lens] Fix mobile view

mbondyra opened the pull request:
## Summary

Fixes https://github.com/elastic/kibana/issues/89077

<img width="164" alt="Screenshot 2021-04-22 at 10 02 55" src="https://user-images.githubusercontent.com/4283304/115678544-eb486200-a351-11eb-9874-9efc68286406.png">


Open dimension panel:
<img width="187" alt="Screenshot 2021-04-22 at 10 18 33" src="https://user-images.githubusercontent.com/4283304/115680723-1c299680-a354-11eb-83f2-2903e967481c.png">

Discussion points (updated after @MichaelMarcialis comment)
1. Heights of the three main sections - suggested solution: accordions for collapsing the sections - seems like a bigger task
2. Order of the sections- keep as it is
3.  Open config panel for mobile should take the whole viewport

What this PR addresses:
- Possibility of scrolling for mobile view
- fixing 3 sections layout to display one after another instead of side by side:
<img width="593" alt="Screenshot 2021-04-22 at 10 33 18" src="https://user-images.githubusercontent.com/4283304/115682862-28aeee80-a356-11eb-9b8b-9fdee8250be6.png">

- the query bar was unnecesarilly fixed as the rest of the header which was leading to very small scrollable space - I aligned it with how visualize solves it

![Apr-22-2021 10-35-20](https://user-images.githubusercontent.com/4283304/115683406-b25ebc00-a356-11eb-97c9-764ae1c4c7e8.gif)

- some overflowing elements were fixed, eg:
<img width="560" alt="Screenshot 2021-04-22 at 10 39 37" src="https://user-images.githubusercontent.com/4283304/115683733-0a95be00-a357-11eb-9f5f-8c6e4bc661c0.png">

- some element are supposed to be displayed side by side were fixed too, eg:

<img width="548" alt="Screenshot 2021-04-22 at 10 51 57" src="https://user-images.githubusercontent.com/4283304/115685713-c3a8c800-a358-11eb-947a-abcaf0e5696d.png">


## Issues I left untouched

I also spot some issues that I didn't address in this PR, as they feel more complicated or they require design input. Still, I think it is worth to merge this PR in this form after review and then iterate on better responsive design. 

<img width="917" alt="Screenshot 2021-04-22 at 11 02 09" src="https://user-images.githubusercontent.com/4283304/115687158-31092880-a35a-11eb-8c2f-79e3066cbbef.png">

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119196/)
* Commit: 5c418a174688078b1c9fcb6834ae1f1d007cb4b8
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-96957/5c418a174688078b1c9fcb6834ae1f1d007cb4b8)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/bfd5b7bda69fde9154b8fc2f955eef5c32f25e33) | [after](https://github.com/elastic/kibana/commit/5c418a174688078b1c9fcb6834ae1f1d007cb4b8) | diff |
| --- | --- | --- | --- |
| `lens` | 942.3KB | 949.7KB | +7.5KB |

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119196/","number":"119196","commit":"5c418a174688078b1c9fcb6834ae1f1d007cb4b8"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120484/)
* Commit: 1ea7ff679cec3614a16b27be1338d2aa2c916a85
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-96957/1ea7ff679cec3614a16b27be1338d2aa2c916a85)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/f5a8405eb97b301021b14f129830975663830555) | [after](https://github.com/elastic/kibana/commit/1ea7ff679cec3614a16b27be1338d2aa2c916a85) | diff |
| --- | --- | --- | --- |
| `lens` | 943.8KB | 951.2KB | +7.5KB |

### History
* :green_heart: [Build #119196](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119196/) succeeded 5c418a174688078b1c9fcb6834ae1f1d007cb4b8

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119196/","number":"119196","commit":"5c418a174688078b1c9fcb6834ae1f1d007cb4b8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120484/","number":"120484","commit":"1ea7ff679cec3614a16b27be1338d2aa2c916a85"}]}
PIPELINE-->

MichaelMarcialis commented:
Hey, @mbondyra! Nice work on this a11y fix. Here's some of my initial thoughts based on your "doubts" list.

> Heights of the three main sections

Regarding the index pattern and subsequent field list content, I believe we can potentially address the height concerns here by collapsing the section down (possibly like an accordion) at the breakpoint threshold where the Lens interface is laid out in a single column. By default, it could be collapsed (in high zoom or small viewport size situations) to avoid taking up an enormous amount of space, but still accessible when the user needs it. This also helps bring the visualization higher on the page.

Regarding the chart layers content, some side conversations have been happening elsewhere, but I think a few folks feel that it would be nice in general in general for chart layers to be able to collapse in some way as well. Doing so would potentially help with the height concerns you raise here (though I would think this to be secondary to the collapsing of the index pattern and field list content, described above).

> Order of the sections (should the datasource panel or workspace be the first one?)

Assuming the above suggestion of collapsing the index pattern and field list content is one that resonates with you as well, my initial instinct would be to keep the order of sections as we currently have them: 1) index pattern and fields list, 2) workspace, 3) layers. I think this order continues to make sense at high zoom or small viewport size situations, as interaction with the fields list still remains the fastest way for a user to craft a visualization (i.e. adding to a workspace). Also, the workspace and layers are somewhat beholden to the fields list, as they both accept the dragging of fields to them, making this order of elements make all the more sense.

> This is the view with open config panel. Is this how we want to solve it?

If possible, I think it would be nice to treat the configuration flyout as a fully overlaid modal experience for, just as the default EUI flyouts currently do (see GIF below). When opened, the configuration flyout overlays the entire experience and focuses the user's attention only on the contents of the flyout. After the user makes their changes, they may close the flyout in order to once again see the content it was covering. Would that be possible?

![flyout-gif](https://user-images.githubusercontent.com/3884767/115300122-988e6080-a12d-11eb-835a-0fa083296b66.gif)

That said, let me know if we want to begin a formal design effort for a fully responsive version of the Lens UI and I can work with the team on how to prioritize.

mbondyra commented:
Hi @MichaelMarcialis I've addressed your comments, please see the updated description of this PR. 

> That said, let me know if we want to begin a formal design effort for a fully responsive version of the Lens UI and I can work with the team on how to prioritize.

Yeah, accordions you mention seem like bigger effort also from developer's side –  let's talk about this during weekly. 

That being said, I still consider this PR an improvement and fixing the accessibility issue that I linked. Could we still merge it in this form and then improve on top, when we start working on the fully responsive version of Lens? (if we decide to do so)

elasticmachine commented:
Pinging @elastic/kibana-app (Team:KibanaApp)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/121665/)
* Commit: a2f5f5247dd257da047e4869b6ebe48f8cb8f5b7
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-96957/a2f5f5247dd257da047e4869b6ebe48f8cb8f5b7)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/121665/execution/node/708/log)
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/121665/execution/node/789/log)

### History
* :green_heart: [Build #120484](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120484/) succeeded 1ea7ff679cec3614a16b27be1338d2aa2c916a85
* :green_heart: [Build #119196](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119196/) succeeded 5c418a174688078b1c9fcb6834ae1f1d007cb4b8

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119196/","number":"119196","commit":"5c418a174688078b1c9fcb6834ae1f1d007cb4b8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120484/","number":"120484","commit":"1ea7ff679cec3614a16b27be1338d2aa2c916a85"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/121665/","number":"121665","commit":"a2f5f5247dd257da047e4869b6ebe48f8cb8f5b7"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/121669/)
* Commit: 1b47c0126b4048bf5c886cce4b1fd70a903101d2
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-96957/1b47c0126b4048bf5c886cce4b1fd70a903101d2)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/705feeb87ba2b1af692ac1e8ca428e4c8ecc55f3) | [after](https://github.com/elastic/kibana/commit/1b47c0126b4048bf5c886cce4b1fd70a903101d2) | diff |
| --- | --- | --- | --- |
| `indexPatternFieldEditor` | 22.1KB | 22.1KB | +11.0B |
| `lens` | 948.5KB | 964.2KB | +15.6KB |
| total |  |  | +15.6KB |

### History
* :broken_heart: [Build #121665](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/121665/) failed a2f5f5247dd257da047e4869b6ebe48f8cb8f5b7
* :green_heart: [Build #120484](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120484/) succeeded 1ea7ff679cec3614a16b27be1338d2aa2c916a85
* :green_heart: [Build #119196](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119196/) succeeded 5c418a174688078b1c9fcb6834ae1f1d007cb4b8

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119196/","number":"119196","commit":"5c418a174688078b1c9fcb6834ae1f1d007cb4b8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120484/","number":"120484","commit":"1ea7ff679cec3614a16b27be1338d2aa2c916a85"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/121665/","number":"121665","commit":"a2f5f5247dd257da047e4869b6ebe48f8cb8f5b7"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/121669/","number":"121669","commit":"1b47c0126b4048bf5c886cce4b1fd70a903101d2"}]}
PIPELINE-->

MichaelMarcialis commented:
> Could we still merge it in this form and then improve on top, when we start working on the fully responsive version of Lens? (if we decide to do so)

I think that's a good approach. What you have here still provides a lot of great fixes. Will review shortly!

myasonik commented:
This PR will bring Lens back to 0 open a11y issues and I'm so excited 🎉

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/121699/)
* Commit: 82032500187c028411e98d273a93c853efb290df
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-96957/82032500187c028411e98d273a93c853efb290df)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/705feeb87ba2b1af692ac1e8ca428e4c8ecc55f3) | [after](https://github.com/elastic/kibana/commit/82032500187c028411e98d273a93c853efb290df) | diff |
| --- | --- | --- | --- |
| `lens` | 948.5KB | 964.2KB | +15.6KB |

### History
* :green_heart: [Build #121669](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/121669/) succeeded 1b47c0126b4048bf5c886cce4b1fd70a903101d2
* :broken_heart: [Build #121665](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/121665/) failed a2f5f5247dd257da047e4869b6ebe48f8cb8f5b7
* :green_heart: [Build #120484](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120484/) succeeded 1ea7ff679cec3614a16b27be1338d2aa2c916a85
* :green_heart: [Build #119196](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119196/) succeeded 5c418a174688078b1c9fcb6834ae1f1d007cb4b8

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119196/","number":"119196","commit":"5c418a174688078b1c9fcb6834ae1f1d007cb4b8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120484/","number":"120484","commit":"1ea7ff679cec3614a16b27be1338d2aa2c916a85"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/121665/","number":"121665","commit":"a2f5f5247dd257da047e4869b6ebe48f8cb8f5b7"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/121669/","number":"121669","commit":"1b47c0126b4048bf5c886cce4b1fd70a903101d2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/121699/","number":"121699","commit":"82032500187c028411e98d273a93c853efb290df"}]}
PIPELINE-->

MichaelMarcialis reviewed (approved):
Thanks for putting this together, Marta. Works as advertised and serves as a great first step in making Lens more responsive.

wylieconlon commented on x-pack/plugins/lens/public/editor_frame_service/editor_frame/config_panel/dimension_container.scss:
```diff
@@ -6,11 +6,15 @@
+  }
+  @include euiBreakpoint('xs', 's', 'm') {
+    @include euiFlyout;
+  }
```
I ran into a similar issue here with the flyout being part of the right side container, but we should really disable scrolling while the flyout is open. I think this is something the EUI components can do for us?

<img width="378" alt="Screen Shot 2021-04-22 at 4 36 32 PM" src="https://user-images.githubusercontent.com/666475/115781934-0aaeb180-a389-11eb-8ad0-9ae09f0281c5.png">

wylieconlon reviewed (approved):
Overall LGTM, tested in Firefox. Found one issue with scrollbars.

mbondyra commented:
@wylieconlon so the problem is that we don't use the eui `EuiFlyout` as for our usecase (we mount it inside the config panel and that's a bit problematic). We use pieces of EuiFlyout - the styling, and internals, like EuiFlyoutHeader and the rest. I was trying to adapt the EuiFlyout and use it, but with no success and I suspect there's not really a good option to do it. Caroline wrote the original code, so I trust this solution and the right usage of eui. 

There's no util that would make a body non-scrollable coming from eui so I just decided to do simple `overflow: hidden` on body when flyout is open. If there's a better way, I'm happy to hear it!

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/121972/)
* Commit: 143dfae623d137ca70c6c1543eba77f57d818c50
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-96957/143dfae623d137ca70c6c1543eba77f57d818c50)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/705feeb87ba2b1af692ac1e8ca428e4c8ecc55f3) | [after](https://github.com/elastic/kibana/commit/143dfae623d137ca70c6c1543eba77f57d818c50) | diff |
| --- | --- | --- | --- |
| `lens` | 948.5KB | 964.6KB | +16.0KB |

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/121972/","number":"121972","commit":"143dfae623d137ca70c6c1543eba77f57d818c50"}]}
PIPELINE-->
